src/coreclr/src/gc/gc.cpp

```diff
@@ -18147,8 +18165,12 @@ void gc_heap::garbage_collect (int n)
             ((settings.pause_mode == pause_interactive) || (settings.pause_mode == pause_sustained_low_latency)))
         {
             keep_bgc_threads_p = TRUE;
-            c_write (settings.concurrent,  TRUE);
+            c_write (settings.concurrent, TRUE);
         }
+
+        memset(&bgc_data_global, 0, sizeof(bgc_data_global));
+        memcpy(&bgc_data_global, &gc_data_global, sizeof(gc_data_global));
```

Comment:
these 2 lines should be in the last if block 'cause if we are not doing a BGC there's no need to do this.